### PR TITLE
Staker wont be able to jump queue anymore.

### LINKED ISSUE #588: Staker can jump the queue at the time of block proposal
Suppose 3 valid blocks have been proposed

```[0,1,2]```

A new block is proposed which has the wrong biggestInfluence and is placed at the following position

```[3,0,1,2]```

Another block is proposed which is a valid block but has an iteration greater than block id 2. It replaces 3 and this is the new ```sortedProposedBlocks``` array

```[4,0,1,2]```

Block id 4 jumps ahead of the rest of the blocks even though it has the highest iteration. Disincentives early block proposals as there can be a chance for a staker to jump the queue incase an invalid block is proposed

### PR DESCRIPTION
**Pull request by SamAg19**
fixes #588

### CODE REVIEW

File: contracts/Core/BlockManager.sol [hunk 1]
```diff
@@ -138,18 +138,6 @@ contract BlockManager is Initializable, BlockStorage, StateManager, BlockManager
         _confirmBlock(epoch - 1, deactivatedCollections, stakerId);
     }
 
-    function disputeBiggestInfluenceProposed(
```
Comment (hrishikeshio) [inline contracts/Core/BlockManager.sol:141]: why is this removed?

**Review by hrishikeshio — CHANGES_REQUESTED** [review]

File: contracts/Core/BlockManager.sol (hunk 1 above)
Comment (SamAg19) [reply contracts/Core/BlockManager.sol:141]: a block with incorrect biggestInfluence can not be a part of the sortedProposedBlockIds. The only way i guess such a block can be a part of this is if all the block proposed have been incorrect.

Comment (hrishikeshio) [reply contracts/Core/BlockManager.sol:141]: thats exactly why this was added.

Comment (SamAg19) [reply contracts/Core/BlockManager.sol]: Added back https://github.com/razor-network/contracts/pull/602/commits/a1be8c7816f8525687ebcdd1da058b39c34158b9

File: contracts/Core/BlockManager.sol [hunk 2]
```diff
@@ -224,14 +224,21 @@ contract BlockManager is Initializable, BlockStorage, StateManager, BlockManager
             return;
         }
 
-        for (uint8 i = 0; i < sortedProposedBlockslength; i++) {
-            // Replace : New Block has better biggest influence
-            if (proposedBlocks[epoch][sortedProposedBlockIds[epoch][i]].biggestInfluence < biggestInfluence) {
-                sortedProposedBlockIds[epoch][i] = blockId;
-                return;
+        if (proposedBlocks[epoch][sortedProposedBlockIds[epoch][0]].biggestInfluence > biggestInfluence) {
+            return;
```
Comment (hrishikeshio) [inline contracts/Core/BlockManager.sol:228]: at line 71, numProposed blocks is increased even though the block was not inserted.

Comment (SamAg19) [reply contracts/Core/BlockManager.sol:228]: Won't that be the case even if a valid block is proposed but length is greater than maxAltBlocks?

Comment (hrishikeshio) [reply contracts/Core/BlockManager.sol:228]: no. in that case a block is actually inserted.

Comment (SamAg19) [reply contracts/Core/BlockManager.sol:228]: But a block is popped off as we will have to maintain the length of maxAltBlocks and numProposedBlocks is also being used as ID for each block that has been proposed.

Comment (hrishikeshio) [reply contracts/Core/BlockManager.sol:228]: why increase numProposed if the block was never inserted. Its as if the block was never proposed.

Comment (SamAg19) [reply contracts/Core/BlockManager.sol:228]: Fixed https://github.com/razor-network/contracts/pull/602/commits/4e178a31d50dd430d69a67a545da1339b41663c7

**Review by hrishikeshio — APPROVED** [review]

**Review by abhishekvispute — APPROVED** [review]
